src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1226,7 +1226,8 @@ async ValueTask<NpgsqlDataReader> ExecuteReaderAsync(CommandBehavior behavior, b
                         // to prevents a dependency on the thread pool (which would also trigger deadlocks).
                         // The WriteBuffer notifies this command when the first buffer flush occurs, so that the
                         // send functions can switch to the special async mode when needed.
-                        var sendTask = NonMultiplexingWriteWrapper(connector, async);
+                        //TODO: pass the token only if the cancellation was not successful
```

Comment:
Any reason we're not flowing here? Are you planning to do this at a later point? This means that if the writing hangs (as opposed to the reading), the user can't cancel.

Reply:
Yes, next pr (it should be the last one). It's already done and is waiting for this one 😏 

If we do pass the token now, there could an undefined behavior (because we're sending the cancellation request when the token is cancelled). And, I would like for it to work as it worked before.

Reply:
Sounds good.